docs/standard/tour.md

```diff
@@ -44,129 +42,79 @@ Microsoft languages that .NET supports include C#, F#, and Visual Basic.
+C#, Visual Basic, and F# support local **type inference**. Type inference means that the compiler will deduce the type of the expression on the left-hand side from the expression on the right-hand side. This doesn't mean that the type safety is broken or avoided. The resulting type **has** a strong type with everything that implies. Let's rewrite the first two lines of the previous example to introduce type inference. You will note that the rest of the example is completely the same.
 
-```cs
-  var dog = Dog.AdoptDog();
-  var pet = (Pet)dog;
-  pet.ActCute();
-  Car car = (Car)dog; // will throw - no relationship between Car and Dog
-  object temp = (object)dog; // legal - a Dog is an object
-  car = (Car)temp; // will throw - the runtime isn't fooled
-  car.Accelerate() // the dog won't like this, nor will the program get this far
+[!code-csharp[TypeSafety](../../samples/csharp/snippets/tour/TypeSafety.csx#L28-L34)]
 
-```
+F# has even further type inference capabilities than method-local type inference found in C# and Visual Basic.  To learn more, check out [Type Inferece](../fsharp/language-reference/type-inference.md).
 
 ## Delegates and lambdas
 
 Delegates are like C++ function pointers, with a big difference that they are type safe. They are a kind of disconnected method within the CLR type system. Regular methods are attached to a class and only directly callable through static or instance calling conventions.
 
 Delegates are used in various APIs and places in the .NET world, especially through lambda expressions, which are a cornerstone of LINQ.
 
-Read more about it in the [Delegates and lambdas](../standard/delegates-lambdas.md) document.
+Read more about it in the [Delegates and lambdas](delegates-lambdas.md) document.
 
-## Generic types (Generics)
+## Generics
 
-Generic types, also commonly called "generics", are a feature that was added in .NET Framework 2.0. In short, generics allow the programmer to introduce a "type parameter" when designing their classes, that will allow the client code (the users of the type) to specify the exact type to use in place of the type parameter.
+Generics are a feature that was added in .NET Framework 2.0. In short, generics allow the programmer to introduce a "type parameter" when designing their classes, that will allow the client code (the users of the type) to specify the exact type to use in place of the type parameter.
 
 Generics were added in order to help programmers implement generic data structures. Before their arrival, in order for a, say, _List_ type to be generic, it would have to work with elements that were of type _object_. This would have various performance as well as semantic problems, not to mention possible subtle runtime errors. The most notorious of the latter is when a data structure contains, for instance, both integers and strings, and an _InvalidCastException_ is thrown on working with the list's members.
```

Comment:
in order to help -> to help

_InvalidCastException_  should be styled as inline code or api link